src/modules/previewpane/powerpreview/settings.cpp

```diff
@@ -33,114 +33,101 @@ namespace PowerPreviewSettings
-    }
-
-    LPCWSTR FileExplorerPreviewSettings::GetSubKey() const
+    LPCWSTR FileExplorerPreviewSettings::GetCLSID() const
     {
-        return preview_handlers_subkey;
+        return this->m_clsid;
     }
 
-    LPCWSTR FileExplorerPreviewSettings::GetCLSID() const
+    std::wstring FileExplorerPreviewSettings::GetRegistryValueData() const
     {
-        return this->m_clsid;
+        return this->m_registryValueData;
     }
 
-    std::wstring FileExplorerPreviewSettings::GetDisplayName() const
+    // Load intital state of the Preview Handler. If no inital state present initialize setting with default value.
+    void FileExplorerPreviewSettings::LoadState(PowerToysSettings::PowerToyValues& settings)
     {
-        return this->m_displayName;
+        auto toggle = settings.get_bool_value(this->GetToggleSettingName());
+        if (toggle)
+        {
+            // If no exisiting setting found leave the default intitialization value.
+            this->UpdateToggleSettingState(*toggle);
+        }
     }
 
-    void FileExplorerPreviewSettings::SetDisplayName(const std::wstring& displayName)
+    // Manage change in state of Preview Handler settings.
+    void FileExplorerPreviewSettings::UpdateState(PowerToysSettings::PowerToyValues& settings, bool enabled)
     {
-        this->m_displayName = displayName;
+        auto toggle = settings.get_bool_value(this->GetToggleSettingName());
+        if (toggle)
+        {
+            auto lastState = this->GetToggleSettingState();
+            if (lastState != *toggle)
+            {
+                this->UpdateToggleSettingState(*toggle);
+
+                // If global setting is enable. Add or remove the preview handler otherwise just change the UI and save the updated config.
+                if (enabled)
+                {
+                    if (lastState)
+                    {
+                        this->DisablePreview();
+                    }
+                    else
+                    {
+                        this->EnablePreview();
+                    }
+                }
+            }
+        }
     }
 
     void FileExplorerPreviewSettings::EnablePreview()
     {
         // Add registry value to enable preview.
-        LONG err = this->m_registryWrapper->SetRegistryValue(HKEY_CURRENT_USER, this->GetSubKey(), this->GetCLSID(), REG_SZ, (LPBYTE)this->GetDisplayName().c_str(), (DWORD)(this->GetDisplayName().length() * sizeof(wchar_t)));
+        LONG err = this->m_registryWrapper->SetRegistryValue(HKEY_CURRENT_USER, preview_handlers_subkey, this->GetCLSID(), REG_SZ, (LPBYTE)this->GetRegistryValueData().c_str(), (DWORD)(this->GetRegistryValueData().length() * sizeof(wchar_t)));
 
         if (err == ERROR_SUCCESS)
         {
-            this->SetState(true);
-            Trace::PreviewHandlerEnabled(true, this->GetDisplayName().c_str());
+            Trace::PreviewHandlerEnabled(true, this->GetToggleSettingName().c_str());
```

Comment:
This will send telemetry event for very enable operation which include every time on the startup of the PowerToys Application. We can send events from `UpdateState` method  since it will be called only when user change the state of any of the previewer from settings UI. And an event when user disable all preview handlers all together from the global setting only in case if the state changed from `enabled` to `disabled` to avoid cases when the PowerToys launch up in `disabled` setting state. @ryanbodrug-microsoft @crutkas Let me know what you think?

Reply:
@udit3333 I think we should make the telemetry event specific to the name of the event as you suggest.  Ie only fire when the state changes.  I would trust your instinct on whatever implementation is the easiest.   An alternative approach might be to check if the registry key exists and only set the reg key and send the telemetry event if it doesn't.

Reply:
Updated the telemetry events to fired on `enabled` and `disabled` of global setting and each time there is a state change in Previewers settings.

Reply:
Successfully validated the expected telemetry events locally.